### DISCUSSION TITLE
DataFlow Refresh Schedule - JSON Sample

### DISCUSSION THREAD

**Discussion by shanerowley-ds** [powerbi-cli/powerbi-cli#53] [Q&A]
I'd love to be able to write a script to assist me creating a DataFlow's Refresh Schedule as it's one of the most tedious tasks. Just struggling to work out how to do this or if anything works.

It appears to be possible according to the following page

https://powerbi-cli.github.io/content/reference-dataflow.html#pbicli-dataflow-refresh-update

But what the "String with the refresh schedule in JSON format." could possible entail eludes me.

There is the PBI API, which I don't want to use, because PBI CLI will defeat the MFA boss for me. MFA via the API is..  scary

https://learn.microsoft.com/en-us/rest/api/power-bi/dataflows/update-refresh-schedule

But I've tried these JSON formats to no luck either.

What would be great is a (working) example on the PBI CLI documentation just like the API has (assuming the API example works - I have naught to get work out it the work yet as I have a lengthy tedious process to setup and connect a Service Principal and all the bs required to overcome the MFA boss first.)

**Comment by liprec** [comment]
Thanks for the feedback and the following command should work with the `schedule.json` file just a 'copy paste' of the body of the first [sample](https://learn.microsoft.com/en-us/rest/api/power-bi/dataflows/update-refresh-schedule#daily-refresh-schedule-example):
` pbicli dataflow refresh update --workspace Demos --dataflow Dataflow --refresh-schedule @schedule.json`

What kind of error(s) are you getting?

But indeed the sample section of the documentation can also contain a `dataflow` part.

**Reply by shanerowley-ds** [reply]
I've been running it in PowerShell. I can't remember all the things I tried last Friday. ChatGPT remembers some, unfortunately, it is unfamiliar with both PowerBI API and CLI, so I had to get creative. iirc, all errors I was experiencing were along the lines of incorrectly guessing what syntax the PBI CLI wanted.

```
$WorkspaceName = "..."
$DataFlowName = "..."

$refreshSchedule = @"
{
"value": {
    "days": [
      "Sunday",
      "Monday",
      "Tuesday",
      "Wednesday",
      "Thursday",
      "Friday",
      "Saturday"
    ],
    "times": [
      "10:00",
      "16:00"
    ],
    "enabled": true,
    "localTimeZoneId": "UTC",
    "notifyOption": "NoNotification"
}
"@

pbicli login
pbicli dataflow refresh update -w $WorkspaceName --dataflow $DataFlowName --refresh-schedule $refreshSchedule 
```

```
node.exe : SyntaxError: Unexpected token v in JSON at position 3
At C:\Users\User\AppData\Roaming\npm\pbicli.ps1:24 char:5
+     & "node$exe"  "$basedir/node_modules/@powerbi-cli/powerbi-cli/bin ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (SyntaxError: Un...N at position 3:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
```
